File: 3-leetcode/02-medium/53. Maximum Subarray.js
// Using Dynamic Programming => T.C: O(n) | S.C: O(1);
const maxSubArray = (nums) => {
    let answer = -Infinity;
    let curr = 0;
    for (let i = 0, len = nums.length; i < len; i++) {
      curr = Math.max(nums[i], curr + nums[i]);
      answer = Math.max(answer, curr);
    }
    return answer;
  };

File: 3-leetcode/02-medium/cpp/53-maximum-subarray.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
  int maxSubArray(vector<int> &nums)
  {
    int _max = INT_MIN;
    int currSum = 0;

    for (int i = 0, n = nums.size(); i < n; i++)
    {
      currSum = max(nums[i], currSum + nums[i]);
      _max = max(_max, currSum);
    }

    return _max;
  }
};